tests/components/vizio/test_media_player.py

```diff
@@ -507,12 +525,29 @@ async def test_setup_with_apps_additional_apps_config(
     caplog: pytest.fixture,
 ) -> None:
     """Test device setup with apps and apps["additional_configs"] in config."""
-    await _test_setup_with_apps(
-        hass,
-        MOCK_TV_WITH_ADDITIONAL_APPS_CONFIG,
-        ADDITIONAL_APP_CONFIG["name"],
-        ADDITIONAL_APP_CONFIG["config"],
-    )
+    async with _cm_for_test_setup_tv_with_apps(
+        hass, MOCK_TV_WITH_ADDITIONAL_APPS_CONFIG, ADDITIONAL_APP_CONFIG["config"],
+    ):
+        attr = hass.states.get(ENTITY_ID).attributes
+        assert attr["source_list"].count(CURRENT_APP) == 1
+        _assert_source_list_with_apps(
+            list(
+                INPUT_LIST_WITH_APPS
+                + APP_LIST
+                + [
+                    app["name"]
+                    for app in MOCK_TV_WITH_ADDITIONAL_APPS_CONFIG[CONF_APPS][
+                        CONF_ADDITIONAL_CONFIGS
+                    ]
+                    if app["name"] not in APP_LIST
+                ]
+            ),
+            attr,
+        )
+        assert ADDITIONAL_APP_CONFIG["name"] in attr["source_list"]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5373293)